src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs

```diff
@@ -98,21 +111,18 @@ where T: AsRef<[u8]>
             .map_err_to_code(ErrorCode::BadBytes, || {
                 format!("Cannot parse value:{} to Date type", v)
             })?;
-        let mut dt = tz
-            .from_local_datetime(&d.and_hms_opt(0, 0, 0).unwrap())
-            .unwrap();
 
         let less_1000 = |dt: DateTime<Tz>| {
             // convert timestamp less than `1000-01-01 00:00:00` to `1000-01-01 00:00:00`
             if dt.year() < 1000 {
-                Ok(tz.from_utc_datetime(
+                tz.from_utc_datetime(
                     &NaiveDate::from_ymd_opt(1000, 1, 1)
                         .unwrap()
                         .and_hms_opt(0, 0, 0)
                         .unwrap(),
-                ))
+                )
             } else {
-                Ok(dt)
+                dt
```

Comment:
maybe we can split read_timestamp_text into 2 local functions: date part and time part, read_date_text only call the former

Reply:
so we nolonger need `.0` or choose DateTime from enum?

Reply:
> maybe we can split read_timestamp_text into 2 local functions: date part and time part, read_date_text only call the former

Can't do this. Support to_date(timestamp)